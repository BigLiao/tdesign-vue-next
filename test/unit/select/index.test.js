import { mount } from '@vue/test-utils';
import Vue from 'vue';
import { Select, OptionGroup, Option } from '@/src/select/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Select', () => {
  // test props api
  describe(':props', () => {
    it(':disabled', () => {
      const wrapper = mount({
        render() {
          return <Select disabled={true}></Select>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':size', () => {
      const wrapper = mount({
        render() {
          return <Select size="large"></Select>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':clearable', () => {
      const wrapper = mount({
        render() {
          return <Select clearable={true}></Select>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':multiple', () => {
      const wrapper = mount({
        render() {
          return <Select multiple={true}></Select>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':placeholder', () => {
      const wrapper = mount({
        render() {
          return <Select placeholder="please select"></Select>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':creatable', () => {
      const wrapper = mount({
        render() {
          return <Select creatable={true}></Select>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':remote', () => {
      const wrapper = mount({
        render() {
          return <Select remote={true}></Select>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':loading', () => {
      const wrapper = mount({
        render() {
          return <Select loading={true}></Select>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':labelInValue', () => {
      const wrapper = mount({
        render() {
          return <Select labelInValue={false}></Select>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':reserveKeyword', () => {
      const wrapper = mount({
        render() {
          return <Select reserveKeyword={false}></Select>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':bordered', () => {      
      const wrapper = mount({
        render() {
          return <Select bordered={true}></Select>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  // test events
  describe('@event', () => {
    // it('visible-change', async () => {
    //   const fn = jest.fn();
    //   const value = 'apple';
    //   const options = [{
    //     label: '苹果apple~~~~',
    //     value: 'apple',
    //   }, {
    //     label: '香蕉banana~~~',
    //     value: 'banana',
    //     disabled: true,
    //   }, {
    //     label: '橘子orange~~~',
    //     value: 'orange',
    //   }]
    //   const wrapper = mount ({
    //     render() {
    //       return <Select onVisibleChange={fn} value={value} options={options}></Select>;
    //     }
    //   });
    //   const selectWrapper = wrapper.find(Select);
    //   selectWrapper.trigger('click');
    //   await Vue.nextTick();
    //   expect(fn).toHaveBeenCalled();
    // });

    it('blur', async () => {
      const fn = jest.fn();
      const wrapper = mount({
        render() {
          return (
            <Select onBlur={fn} filterable={true}>
              <Option value={1} label={1}>1</Option>
            </Select>
          );
        },
      });
      const selectWrapper = wrapper.find(Select);
      const inputElemWrapper = wrapper.find('input');
      selectWrapper.vm.focus();
      selectWrapper.vm.blur();
      inputElemWrapper.trigger('blur');
      await Vue.nextTick();
      expect(selectWrapper.emitted().blur).toBeTruthy();
    });

    it('focus', async () => {
      const fn = jest.fn();
      const wrapper = mount({
        render() {
          return (
            <Select onFocus={fn} filterable={true}>
              <Option value={1} label={1}>1</Option>
            </Select>
          );
        },
      });
      const selectWrapper = wrapper.find(Select);
      const inputElemWrapper = wrapper.find('input');
      selectWrapper.vm.focus();
      inputElemWrapper.trigger('focus');
      await Vue.nextTick();
      expect(selectWrapper.emitted().focus).toBeTruthy();
    });

    it('clear', async () => {
        const fn = jest.fn();
        let value = 'apple';
        const options = [{
          label: '苹果apple~~~~',
          value: 'apple',
        }, {
          label: '香蕉banana~~~',
          value: 'banana',
          disabled: true,
        }, {
          label: '橘子orange~~~',
          value: 'orange',
        }]
        const wrapper = mount ({
          render() {
            return <Select clearable={true} v-model={value} options={options}></Select>;
          }
        });
        const selectWrapper = wrapper.find(Select);
        const closeIconWrapper = wrapper.find('.t-icon-close')
        closeIconWrapper.trigger('click')
        await Vue.nextTick();
        expect(selectWrapper.emitted().clear).toBeTruthy();    
      });

      it('remove', async () => {
        const fn = jest.fn();
        let value = ['apple'];
        const options = [{
          label: '苹果apple~~~~',
          value: 'apple',
        }, {
          label: '香蕉banana~~~',
          value: 'banana',
          disabled: true,
        }, {
          label: '橘子orange~~~',
          value: 'orange',
        }]
        const wrapper = mount ({
          render() {
            return <Select multiple={true} v-model={value} options={options}></Select>;
          }
        });
        const selectWrapper = wrapper.find(Select);
        const closeIconWrapper = wrapper.find('.t-icon-close')
        closeIconWrapper.trigger('click')
        await Vue.nextTick();
        expect(selectWrapper.emitted().remove).toBeTruthy();
      });
  });
});

describe('Select Option', () => {
  // test props api
  describe(':props', () => {
    it(':value', () => {
      let value = '1'
      const wrapper = mount({
        render() {
          return (
            <Select v-model={value}>
              <Option value={'1'} label={'1'}></Option>
            </Select>          
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });   
    it(':label', () => {
      let value = '1'
      const wrapper = mount({
        render() {
          return (
            <Select v-model={value}>
              <Option value={'1'} label={'1'}></Option>
            </Select>          
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });   
    it(':disabled', () => {
      let value = '1'
      const wrapper = mount({
        render() {
          return (
            <Select v-model={value}>
              <Option value={'1'} label={'1'} disabled={true}></Option>
            </Select>          
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });   
  });
});

describe('Select OptionGroup', () => {
  // test props api
  describe(':props', () => {
    it(':value', () => {
      let value = '1'
      const wrapper = mount({
        render() {
          return (
            <Select v-model={value}>
              <OptionGroup label={'num'}>
                <Option value={'1'} label={'1'}></Option>
              </OptionGroup>
              <OptionGroup label={'abc'}>
                <Option value={'a'} label={'a'}></Option>
              </OptionGroup>
            </Select>          
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    }); 
  });
});

