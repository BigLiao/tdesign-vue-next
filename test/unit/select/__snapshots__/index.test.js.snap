// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select :props :bordered 1`] = `
<div
  class="t-select-wrap"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              <!---->
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <span
        class="t-select-placeholder"
      >
         
        
      </span>
      
      
      <!---->
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select :props :clearable 1`] = `
<div
  class="t-select-wrap"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              <!---->
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <span
        class="t-select-placeholder"
      >
         
        
      </span>
      
      
      <!---->
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select :props :creatable 1`] = `
<div
  class="t-select-wrap"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              <!---->
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <span
        class="t-select-placeholder"
      >
         
        
      </span>
      
      
      <!---->
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select :props :disabled 1`] = `
<div
  class="t-select-wrap"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-is-disabled t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              <!---->
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-is-disabled t-size-m"
    >
      <!---->
      <span
        class="t-select-placeholder"
      >
         
        
      </span>
      
      
      <!---->
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select :props :labelInValue 1`] = `
<div
  class="t-select-wrap"
  labelinvalue="false"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              <!---->
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <span
        class="t-select-placeholder"
      >
         
        
      </span>
      
      
      <!---->
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select :props :loading 1`] = `
<div
  class="t-select-wrap"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <li
              class="t-select-loading-tips t-size-m"
            >
              加载中
            </li>
            <!---->
            <span
              style="display: none;"
            >
              <!---->
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <span
        class="t-select-placeholder"
      >
         
        
      </span>
      
      
      <!---->
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select :props :multiple 1`] = `
<div
  class="t-select-wrap"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-is-selected-multiple t-size-m t-select-create-option-special"
                title=""
              >
                <label
                  class="t-checkbox undefined"
                  modelvalue="false"
                >
                  <input
                    class="t-checkbox__former"
                    type="checkbox"
                  />
                  <span
                    class="t-checkbox__input"
                  />
                  <span
                    class="t-checkbox__label"
                  >
                    
                    
                    
                  </span>
                </label>
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              <!---->
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <span
        class="t-select-placeholder"
      >
         
        
      </span>
      
      
      <!---->
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select :props :placeholder 1`] = `
<div
  class="t-select-wrap"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              <!---->
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <span
        class="t-select-placeholder"
      >
         
        please select
      </span>
      
      
      <!---->
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select :props :remote 1`] = `
<div
  class="t-select-wrap"
  remote="true"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              <!---->
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <span
        class="t-select-placeholder"
      >
         
        
      </span>
      
      
      <!---->
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select :props :reserveKeyword 1`] = `
<div
  class="t-select-wrap"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              <!---->
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <span
        class="t-select-placeholder"
      >
         
        
      </span>
      
      
      <!---->
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select :props :size 1`] = `
<div
  class="t-select-wrap"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-l t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-l"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              <!---->
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-l"
    >
      <!---->
      <span
        class="t-select-placeholder"
      >
         
        
      </span>
      
      
      <!---->
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select Option :props :disabled 1`] = `
<div
  class="t-select-wrap"
  modelvalue="1"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              
              <li
                class="t-select-option t-is-disabled t-is-selected t-size-m"
                title="1"
              >
                1
              </li>
              
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <!---->
      
      
      <span
        class="t-select-selectedSingle"
        title="1"
      >
        1
      </span>
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select Option :props :label 1`] = `
<div
  class="t-select-wrap"
  modelvalue="1"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              
              <li
                class="t-select-option t-is-selected t-size-m"
                title="1"
              >
                1
              </li>
              
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <!---->
      
      
      <span
        class="t-select-selectedSingle"
        title="1"
      >
        1
      </span>
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select Option :props :value 1`] = `
<div
  class="t-select-wrap"
  modelvalue="1"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              
              <li
                class="t-select-option t-is-selected t-size-m"
                title="1"
              >
                1
              </li>
              
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <!---->
      
      
      <span
        class="t-select-selectedSingle"
        title="1"
      >
        1
      </span>
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Select OptionGroup :props :value 1`] = `
<div
  class="t-select-wrap"
  modelvalue="1"
>
  <div
    class="t-popup-reference t-select-popup-reference"
  >
    <transition-stub>
      <div
        aria-hidden="true"
        class="t-popup"
        role="tooltip"
        style="display: none;"
      >
        <div
          class="t-popup-content t-select-dropdown narrow-scrollbar"
          style=""
        >
          
          <div>
            <ul
              class="t-select-create-option"
              style="display: none;"
            >
              <li
                class="t-select-option t-size-m t-select-create-option-special"
                title=""
              >
                
              </li>
            </ul>
            <!---->
            <li
              class="t-select-empty t-size-m"
            >
              暂无数据
            </li>
            <span
              style="display: none;"
            >
              
              <li
                class="t-option-group t-size-m"
              >
                <ul
                  class="t-option-group-header"
                >
                  num
                </ul>
                <ul>
                  
                  <li
                    class="t-select-option t-is-selected t-size-m"
                    title="1"
                  >
                    1
                  </li>
                  
                </ul>
              </li>
              <li
                class="t-option-group t-size-m"
              >
                <ul
                  class="t-option-group-header"
                >
                  abc
                </ul>
                <ul>
                  
                  <li
                    class="t-select-option t-size-m"
                    title="a"
                  >
                    a
                  </li>
                  
                </ul>
              </li>
              
            </span>
          </div>
          
          <!---->
        </div>
      </div>
    </transition-stub>
    
    <div
      class="t-select t-size-m"
    >
      <!---->
      <!---->
      
      
      <span
        class="t-select-selectedSingle"
        title="1"
      >
        1
      </span>
      <!---->
      <svg
        class="t-select-right-icon t-fake-arrow"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
          stroke="black"
          stroke-opacity="0.9"
          stroke-width="1.3"
        />
      </svg>
      <!---->
      <!---->
    </div>
    
  </div>
</div>
`;
